### ISSUE TITLE
Cast of annotated type is incorrectly reporting unnecessary cast

### ISSUE THREAD

**Issue by runfalk** [microsoft/pyright#7294]
**Describe the bug**
I was working with the `pydantic.TypeAdapter` class which is typed around a generic `type[T]`. I ran into the issue described in #7238 and https://github.com/pydantic/pydantic/issues/8735.

Thought it's not pretty I thought `t.cast(ActualType, AnnotatedType)` was a reasonable enough workaround for this use case. However, doing  this results in `reportUnnecessaryCast`, which seems odd since the value is not treated that way when used. It doesn't matter if I do it when defining the type alias or when using the type. There is an argument to be made that they are the same when the type alias is defined, but not when used as a value.

The best workaround I've found for now is:

```
TypeAdapter[int](t.cast(t.Any, AnnotatedInt))
```

**Code + Output**
```
$ poetry run pyright --version
pyright 1.1.350
```

```python
import typing as t
AnnotatedInt = t.cast(type[int], t.Annotated[int, ...])
```

```
error: Unnecessary "cast" call; type is already "type[int]" (reportUnnecessaryCast)
```

**Comment by erictraut** [comment]
Yes, this is unintended behavior, so I consider it a bug.

It will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.351, which I just published.
